storage/innobase/trx/trx0roll.cc

```diff
@@ -128,6 +128,7 @@ trx_rollback_to_savepoint_low(
 	} else {
 		trx->lock.que_state = TRX_QUE_RUNNING;
 		MONITOR_INC(MONITOR_TRX_ROLLBACK_SAVEPOINT);
+		trx->vers_update_trt = savept->vers_update_trt;
```

Comment:
Why do we need both fields? And what exactly is the purpose of this field?

Reply:
That's how savepoint works: it stores current data and restores it on rollback to savepoint. `vers_update_trt` notifies that update of TRT is required.